hal/ia32/pci.c

```diff
@@ -122,21 +108,19 @@ int hal_pciSetBusmaster(pci_dev_t *dev, u8 enable)
 int hal_pciGetDevice(pci_id_t *id, pci_dev_t *dev, void *caps)
 {
 	spinlock_ctx_t sc;
-	unsigned char b, d, f, i;
+	unsigned int b, d, f, i;
```

Comment:
Why change to `unsigned int`? `_hal_pciGet` still gets arguments of `u8` type.

Reply:
The `b` index in `for (b = dev->bus; b < 256; b++)` being unsigned char caused infinite loop if no device was detected. Making it unsigned int solves the issue.